window/src/os/wayland/drag_and_drop.rs

```diff
@@ -0,0 +1,116 @@
+use crate::connection::ConnectionOps;
+use crate::wayland::{read_pipe_with_timeout, WaylandConnection};
+use filedescriptor::Pipe;
+use smithay_client_toolkit as toolkit;
+use std::os::unix::io::AsRawFd;
+use toolkit::reexports::client::protocol::wl_data_device::Event as DataDeviceEvent;
+use toolkit::reexports::client::protocol::wl_data_offer::WlDataOffer;
+use url::Url;
+use wayland_client::protocol::wl_data_device_manager::DndAction;
+
+#[derive(Default)]
+pub struct DragAndDrop {
+    offer: Option<(u32, WlDataOffer)>, // surface id, offer
+}
+
+pub const URI_MIME_TYPE: &str = "text/uri-list";
+
+impl DragAndDrop {
+    pub fn handle_data_event(&mut self, event: DataDeviceEvent) {
+        match event {
+            DataDeviceEvent::Enter {
+                serial,
+                surface,
+                id,
+                ..
+            } => {
+                if let Some(offer) = id {
+                    offer.accept(serial, Some(URI_MIME_TYPE.to_string()));
+                    offer.set_actions(
+                        DndAction::None | DndAction::Copy | DndAction::Move,
```

Comment:
Does it make sense to allow `Move` here? It feels like that might be a bit of lie!  Perhaps also true of `Copy`.  It probably doesn't really matter, I'm just curious if this is necessary or if there are negative consequences

Reply:
Huh, to be honest, I am not too sure either. For example my file manager only provided `Copy` and `Move`, but no `None` action. So I just thought let's keep the restrictions relaxed. I would say `Copy` is a must, but I'm not too sure about `Move`. If you think I should remove it, I can do that.

Reply:
Done